audio/out/ao_wasapi.c

```diff
@@ -406,20 +406,28 @@ static int thread_control(struct ao *ao, enum aocontrol cmd, void *arg)
         if (!state->pSessionControl)
             return CONTROL_FALSE;
 
-        wchar_t *title = mp_from_utf8(NULL, (char*)arg);
-        wchar_t *tmp = NULL;
-        // There is a weird race condition in the IAudioSessionControl itself --
-        // it seems that *sometimes* the SetDisplayName does not take effect and
-        // it still shows the old title. Use this loop to insist until it works.
-        do {
-            IAudioSessionControl_SetDisplayName(state->pSessionControl, title, NULL);
-
-            SAFE_DESTROY(tmp, CoTaskMemFree(tmp));
-            IAudioSessionControl_GetDisplayName(state->pSessionControl, &tmp);
-        } while (wcscmp(title, tmp));
-        SAFE_DESTROY(tmp, CoTaskMemFree(tmp));
+        wchar_t *title = mp_from_utf8(NULL, (const char *)arg);
+        HRESULT hr = IAudioSessionControl_SetDisplayName(state->pSessionControl,
+                                                         title,NULL);
         talloc_free(title);
-        return CONTROL_OK;
+
+        if (SUCCEEDED(hr))
+            return CONTROL_OK;
+
+        MP_WARN(ao, "Error setting audio session name: %s\n",
+                mp_HRESULT_to_str(hr));
+
+        assert(ao->client_name);
+        if (!ao->client_name)
```

Comment:
mpv has assertions in release builds so this is actually dead code

Reply:
what if someone defined `NDEBUG`?  I would still check it, just because it doesn't harm.